src/components/icon/index.tsx

```diff
@@ -19,7 +19,7 @@ export default function Icon({ name, data, onClose }: IconProps) {
 
   return (
     <button
-      onClick={() => e(data.url, data.text, data.title) && onClose()}
+      onClick={() => e(decodeURIComponent(data.url || ""), data.text, data.title) && onClose()}
```

Comment:
empty condition will be taken care initially so no need of `|| ""`

also shouldn't it be `encodeURIComponent` since we want to pass it encoded to our UI?

Reply:
Since in native web share sheet we will pass non encoded URL

Reply:
Yea my bad, must have been the sleep deprivation 😅
About the `|| ""`, I added this because `data.url` is optional in `ShareData` and the ts compiler was throwing compile error. that can be solved by making `url` non-optional